### ISSUE TITLE
Add the ability to 'define' keywords

### ISSUE THREAD

**Issue by danielcompton** [cursive-ide/cursive#1549]
Here are two use cases where it would be helpful to be able to 'define' a keyword and navigate to where it was 'defined':
1. specs: `s/def` and friends
2. re-frame: In re-frame you register effects and subscriptions to keywords (e.g. `:app/do-something`).

``` clj
(reg-event-fx
  :app/do-something
  (fn [{:keys [db]} [_ val]]
    (do something)
    ))
```

When navigating through re-frame code, we often want to navigate to where this keyword was registered. Find usages works, but shows you all usages of the keyword, not where it was registered. I imagine spec will have the same issues?

**Comment by superstructor** [comment]
This would be a great feature to have. Interested to know how Cursive will tell the difference between the definition `(reg-event-fx :app/do-something ...` and the usage `(dispatch [:app/do-something])` where `:app/` is a synthetic namespace ?

**Comment by souenzzo** [comment]
IMHO it should be an knowledge about `re-frame`/`spec` code...
Maybe "Case contains `re-frame.core` on imported namespaces, than keys after `reg-event-fx` are `definitions`" or something like...

**Comment by cursive-ide** [comment]
This is fixed in the next build in #1470.
